### PR TITLE
feat: compile executable

### PR DESCRIPTION
**Pull request by FroyaTheHen**
commit-id:4510e5a4

### CODE REVIEW

File: scarb/src/compiler/compilers/executable.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    fn compile(
+        &self,
+        unit: CairoCompilationUnit,
+        db: &mut RootDatabase,
+        ws: &Workspace<'_>,
+    ) -> Result<()> {
+        let target_dir = unit.target_dir(ws);
+        let main_crate_ids = collect_main_crate_ids(&unit, db);
+        let compiler_config = build_compiler_config(db, &unit, &main_crate_ids, ws);
+        let _ = trace_span!("compile_executable").enter();
```
Comment (maciektr) [inline scarb/src/compiler/compilers/executable.rs:26]:
```suggestion
        let _guard = trace_span!("compile_executable").enter();
```
Would you mind changing it here and in other places? TIL `let _ =` might drop right away :D

Comment (maciektr) [reply scarb/src/compiler/compilers/executable.rs]: Unresolving, as it's still not changed :D `let _ = trace_span`

Comment (FroyaTheHen) [reply scarb/src/compiler/compilers/executable.rs]: sorry, my bad 🙈 
`let _guard = trace_span!("compile_executable").enter();` still raises `temporary value dropped while borrowed [E0716]`. 
How about 👇 
```
        let span = trace_span!("compile_executable");
        let executable = {
            let _guard = span.enter();
            Executable::new(
                cairo_lang_executable::compile::compile_executable_in_prepared_db(
                    db,
                    None,
                    main_crate_ids,
                    compiler_config.diagnostics_reporter,
                )?,
            )
        };
```
??
Ofc, I will change other places as well, just want to consult it first with you :)

Comment (Draggu) [reply scarb/src/compiler/compilers/executable.rs]: https://docs.rs/tracing/latest/tracing/struct.Span.html#method.in_scope

Comment (maciektr) [reply scarb/src/compiler/compilers/executable.rs]: ^ Both are fine imho. `in_scope` is only unavoidable when working with asyncs.

File: scarb/src/compiler/compilers/lib.rs [hunk 2]
```diff
@@ -114,8 +116,9 @@ impl Compiler for LibCompiler {
                 .context("failed calculating Sierra variables")?
             };
 
+            let span = trace_span!("compile_casm");
             let cairo_program = {
-                let _ = trace_span!("compile_casm").enter();
+                let _ = span.enter();
```
Comment (maciektr) [inline scarb/src/compiler/compilers/lib.rs:121]:
```suggestion
                let _guard = span.enter();
```

**Review by maciektr — APPROVED** [review]

**Review by maciektr — APPROVED** [review]

**Pull request merged**